crates/task-impls/src/quorum_vote.rs

```diff
@@ -0,0 +1,359 @@
+    _disperse: Proposal<TYPES, VidDisperse<TYPES>>,
+    _event_sender: Sender<HotShotEvent<TYPES>>,
+) -> bool {
+    true
+}
+
+/// The state for the quorum vote task.
+///
+/// Contains all of the information for the quorum vote.
+pub struct QuorumVoteTaskState<TYPES: NodeType, I: NodeImplementation<TYPES>> {
+    /// Smallest potential view number that will be voted for.
+    pub next_vote_view: TYPES::Time,
+
+    /// Table for the in-progress vote dependencies.
+    pub vote_dependencies: HashMap<TYPES::Time, JoinHandle<()>>,
+
+    /// Event receiver.
+    pub event_receiver: Receiver<HotShotEvent<TYPES>>,
+
+    /// Network for all nodes
+    pub quorum_network: Arc<I::QuorumNetwork>,
+
+    /// Network for DA committee
+    pub committee_network: Arc<I::CommitteeNetwork>,
+
+    /// Output events to application
+    pub output_event_stream: async_broadcast::Sender<Event<TYPES>>,
+
+    /// The node's id
+    pub id: u64,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> QuorumVoteTaskState<TYPES, I> {
+    /// Create an event dependency.
+    #[instrument(skip_all, fields(id = self.id, next_vote_view = *self.next_vote_view), name = "Quorum vote validate VID", level = "error")]
+    fn create_event_dependency(
+        &self,
+        dependency_type: VoteDependency,
+        view_number: TYPES::Time,
+        event_sender: Sender<HotShotEvent<TYPES>>,
+    ) -> EventDependency<HotShotEvent<TYPES>> {
+        EventDependency::new(
+            self.event_receiver.clone(),
+            Box::new(move |event| match dependency_type {
+                VoteDependency::QuorumProposal => {
+                    if let HotShotEvent::QuorumProposalRecv(proposal, _) = event {
+                        if proposal.data.view_number != view_number {
+                            return false;
+                        }
+                        return validate_quorum_proposal(proposal.clone(), event_sender.clone());
+                    }
+                    false
+                }
+                VoteDependency::Dac => {
+                    if let HotShotEvent::DACRecv(cert) = event {
+                        if cert.view_number != view_number {
+                            return false;
+                        }
+                        return validate_dac(cert.clone(), event_sender.clone());
+                    }
+                    false
+                }
+                VoteDependency::Vid => {
+                    if let HotShotEvent::VidDisperseRecv(disperse, _) = event {
+                        if disperse.data.view_number != view_number {
+                            return false;
+                        }
+                        return validate_vid(disperse.clone(), event_sender.clone());
+                    }
+                    false
+                }
+            }),
+        )
+    }
+
+    /// Create an [`AndDependency`] combining three [`EventDependency`]s.
+    fn create_vote_dependency(
```

Comment:
How will the vote dependency (`AndDependency`) get access to the event which triggered this function to be called?

Reply:
I've replaced the `create_vote_dependency` function with `create_dependency_task_if_new`, and it will be called every time we receive an event and want to create a new `DependencyTask`.